src/cargo/ops/cargo_run.rs

```diff
@@ -104,6 +104,10 @@ pub fn run(
         process.display_env_vars();
     }
 
+    if let Some(client) = config.jobserver_from_env() {
+        process.inherit_jobserver(client);
```

Comment:
From the 99% case of `cargo run` these semantics feel a bit odd.

I guess to break this down to more concrete points to overcome the oddity
- Can this change negatively impact anyone?
- Would it be a breaking change if we felt we needed to remove this in the future?

Reply:
> Can this change negatively impact anyone?

Unless you're using `make` with `-j` arg, otherwise there is no jobserver you can inherit from the environment. So 99% case of `cargo run` won't inherit this.

> Would it be a breaking change if we felt we needed to remove this in the future?

Yes for sure. for those going to depend on this change like cargo-miri.

Reply:
Overall, this feels too specialized for me and makes me wonder if we are solving the right problem.